services/business/services.go

```diff
@@ -83,3 +83,19 @@ func (in *SvcService) GetService(namespace, service, interval string) (*models.S
 	s.SetServiceDetails(serviceDetails, istioDetails, prometheusDetails)
 	return &s, nil
 }
+
+// GetApps returns a list of "app" label values used for the Deployments covered by this service
```

Comment:
This is still a big question for me, I may not yet have seen the light, do we need to be dealing with services and aggregating across apps?  Also, is it typical for a Service to have deployments with different app labels?  I'd like to have a call where someone clearly explains the "app" labeling to me.

Reply:
So this is called from Service Details pages, so we really start from a service and nothing else. This is the only way I can think of, to get the related metrics.

As for the possibility to have multiple apps: I cannot say if it could be a common use case or if it's just a weird corner case. But it's technically possible, apps cardinality is [0,n], so it dictates to manage the case.